src/Adapter/HookManager.php

```diff
@@ -80,6 +81,9 @@ public function exec(
             } catch (\Exception $e) {
                 $logger = ServiceLocator::get('\\PrestaShop\\PrestaShop\\Adapter\\LegacyLogger');
                 $logger->error(sprintf('Exception on hook %s for module %s. %s', $hook_name, $id_module, $e->getMessage()), ['object_type' => 'Module', 'object_id' => $id_module, 'allow_duplicate' => true]);
+                if ((new Environment())->isDebug()) {
```

Comment:
Isn't it possible to pass it to constructor ?

Like here : https://github.com/PrestaShop/PrestaShop/blob/ad181f7fb89dea67dfebba779a6440854baa6d5b/src/PrestaShopBundle/Resources/config/services/adapter/data_configuration.yml#L82

Reply:
There are two possibilities actually, either inject the `prestashop.adapter.environment` service into the HookManager or the boolean directly Both are valid choices

Reply:
By the way @lmeyer1 if you're not used to Symfony services or dependency injection here's a link to the Symfony documentation that might help you https://symfony.com/doc/3.4/service_container.html#explicitly-configuring-services-and-arguments

Reply:
You can't use HookManager as a service, it's instantiate manually. But you're right, since Symfony Container is used in this file, you can reach the `prestashop.adapter.environment` service instead of creating a new instance of Environement :+1:

Reply:
@PierreRambaud 
Do you mean using `ServiceLocator` ? Like

    $env = ServiceLocator::get('\\PrestaShop\\PrestaShop\\Adapter\\Environment');